book/src/htlc.md

```diff
@@ -0,0 +1,105 @@
+# Hashed Time-Locked Contract Transactions
+
+A Hashed Time-Locked Contract is a transaction that allows one party of a
+transaction to spend tokens by forced-disclosure of the preimage of a message
+digest, and allows the counterparty to spend the tokens after a predefined
+lockout period.
+
+Another way to look at HTLC is that it is a protocol to eliminate [counterparty
+risk] where one side of a trade could default on their obligation.
+
+The key requirements for HTLC is that a ledger supports time-locked transactions,
+meaning a transaction that won't be executed until at a later point in time. And
+conditional transactions with message digest instructions.
+
+## Atomic Swaps
+
+One use case of HTLC is atomic swaps. It solves the problem of (at least) two
+parties wanting to exchange their DLT tokens or coins without having to trust
+a third party or a centralized exchange.
+
+A solution to this problem was [first suggested by Tier Nolan on the Bitcoin forum]
+where he described a risk-free within a time bound interactive protocol to exchange
```

Comment:
Is there a word missing between "risk-free" and "within"?

Reply:
rewritten.